### ISSUE TITLE
Add `ignoreAtRules: []` to `nesting-selector-no-missing-scoping-root`

### ISSUE THREAD

**Issue by saitho** [stylelint/stylelint#8728]
### What minimal example or steps are needed to reproduce the bug?

```css
@mixin test() {
	&:before {
		content: "a";
	}

	&:hover:before {
		content: "b";
	}
}
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "extends": [
    "stylelint-config-recommended-scss"
  ]
}
```

### How did you run Stylelint?

[Demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcIACBbAlgDzQOwAQBcYBnfACgEpdgAdPXAMngCMYAzCAJxituvz8jZCQ+LmogAhuIDctXLgC+vbH0YALCADcYHZm07caK-vkHD8o8Uxm98S7ApAAaEKzQAbGADkJKOIhgMXwAHTwA6MCIiZ3AIbDcAcwQQI3lxQOEoInFRAG05eTEQEgBPT3ccfABaQUSqrkgUP2xYKCqiSOyQAoBdWnsY2rQEgDFOFAl8ZIArIjiY2GDoxFSi0vLKnKL3SeJ+ZwLxdZgKoRq4uoaIJpgWmDaOqK3xHcIScX6QBSA)

### Which Stylelint-related dependencies are you using?

```json
{
  "stylelint": "latest",
  "stylelint-config-recommended-scss": "latest"
}
```

### What did you expect to happen?

I expect `&` to work in mixins, as the scope is present when mixin is resolved via `@include`

### What actually happened?

```
2:3-4 error Unexpected missing scoping root (nesting-selector-no-missing-scoping-root)
6:2-3 error Unexpected missing scoping root (nesting-selector-no-missing-scoping-root)
```

**Comment by jeddy3** [comment]
@saitho Thanks for the report and reproducible demo.

We typically [add secondary options](https://stylelint.io/developer-guide/rules#write-the-rule) to help language extension communities, like the SCSS one, use our built-in rules.

For example, like [how they do](https://github.com/stylelint-scss/stylelint-config-recommended-scss/blob/c55487214c69b5b899b38076b964e14c3c9bcfb1/index.js#L22-L27) with the `no-invalid-position-at-import-rule` rule:

```json
{
  "rules": {
    "no-invalid-position-at-import-rule": [ true, {
      "ignoreAtRules": ["use'", "forward"],
    }]
  }
}
```

Let's add an `ignoreAtRules: []` option to `nesting-selector-no-missing-scoping-root` so that the SCSS community can configure it in their shared config:

```json
{
  "rules": {
    "nesting-selector-no-missing-scoping-root": [ true, {
      "ignoreAtRules": ["mixin"],
    }]
  }
}
```

I've labelled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to add a new option](https://stylelint.io/developer-guide/rules#add-an-option-to-a-rule) in the Developer guide.

Let's use `ignoreAtRules: ["--foo", "/^--bar-/"]` in the tests and docs to be consistent with our conventions.

**Comment by karlhorky** [comment]
@jeddy3 I opened a PR for `ignoreAtRules` in `nesting-selector-no-missing-scoping-root` here:

- #8743

**Comment by karlhorky** [comment]
Thanks for the review, merge and release of PR #8743! 🙌

I can confirm that [`stylelint@16.24.0`](https://github.com/stylelint/stylelint/releases/tag/16.24.0) works with the config below for [Tailwind CSS `@utility` at-rules](https://tailwindcss.com/docs/adding-custom-styles#simple-utilities) 🎉

`stylelint.config.js`

```ts
/** @type {import('stylelint').Config} */
const config = {
  rules: {
    'nesting-selector-no-missing-scoping-root': [
      true,
      {
        ignoreAtRules: ['utility'],
      },
    ],
  },
};

export default config;
```

`app/globals.css`

```css
@utility content-body {
  /* No longer reported as a problem */
  & p {
    @apply mb-4;
  }
}
```
